src/core/meshcop/timestamp.hpp

```diff
@@ -36,13 +36,11 @@
 
 #include "openthread-core-config.h"
 
-#include <string.h>
-
 #include <openthread/dataset.h>
-#include <openthread/platform/toolchain.h>
 
 #include "common/clearable.hpp"
 #include "common/encoding.hpp"
+#include "common/error.hpp"
```

Comment:
I found `#include "common/error.hpp"` is added in a few places while `<openthread/platform/toolchain.h>` is not added in those places. Not sure if this is intentional. But I think we'd better explicitly include  `<openthread/platform/toolchain.h>` if the sources needs some definitions from it. Thoughts?